dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -23,12 +23,134 @@ def self.extract_parent_emails
     source_sql = <<~SQL
       SELECT parent_email, MAX(updated_at) AS updated_at
       FROM users
+      WHERE parent_email > ''
       GROUP BY parent_email
     SQL
     query = get_extraction_query(source_sql, 'parent_email', [], true, 'dashboard.users.parent_email')
     ContactRollupsV2.execute_query_in_transaction(query)
   end
 
+  def self.extract_user_geos
+    teacher_query = <<~SQL
+      SELECT id, email
+      FROM users
+      WHERE user_type = 'teacher' AND email > ''
+    SQL
+
+    # An user can have many user_geos records. user_geos records starts with only NULL
+    # values until a cronjob runs, does IP-to-address lookup, and update them later.
+    teacher_geos_query = <<~SQL
+      SELECT
+        teachers.email,
+        user_geos.city, user_geos.state, user_geos.postal_code, user_geos.country,
+        MAX(user_geos.updated_at) as updated_at
+      FROM (#{teacher_query}) AS teachers
+      JOIN user_geos
+      ON teachers.id = user_geos.user_id
+      GROUP BY email, city, state, postal_code, country
+    SQL
+
+    extraction_query = get_extraction_query(
+      teacher_geos_query,
+      'email',
+      %w(city state postal_code country),
+      true,
+      'dashboard.user_geos'
+    )
+    ContactRollupsV2.execute_query_in_transaction(extraction_query)
+  end
+
+  def self.extract_pd_enrollments
+    enrollment_email_query = <<~SQL
+      SELECT email, MAX(updated_at) AS updated_at
+      FROM pd_enrollments
+      WHERE email > ''
+      GROUP BY email
+    SQL
+
+    extraction_query = get_extraction_query(
```

Comment:
Not sure it matters, but you don't need to name this since it's a single table query right?

Reply:
Ah you have to name it because there's a subquery b/c email is unique.

Reply:
It seems like the use of subqueries will turn out to be the rule, not the exception. Wonder if we could make the method signature shorter in that case.

Reply:
lol didn't see your 3rd comment before I post mine.